app/src/main/java/org/fossasia/openevent/general/auth/RequestAuthenticator.kt

```diff
@@ -2,11 +2,23 @@ package org.fossasia.openevent.general.auth
 
 import okhttp3.Interceptor
 import okhttp3.Response
+import org.fossasia.openevent.general.HEADER_TYPE
+import org.fossasia.openevent.general.JWT_ACCESS_TOKEN
+import org.fossasia.openevent.general.JWT_REFRESH_TOKEN
+import org.fossasia.openevent.general.data.Preference
 
-class RequestAuthenticator(private val authHolder: AuthHolder) : Interceptor {
+class RequestAuthenticator(
+    private val authHolder: AuthHolder,
+    private val preference: Preference
+) : Interceptor {
 
     override fun intercept(chain: Interceptor.Chain): Response {
-        val authorization = authHolder.getAuthorization()
+        val headerType = preference.getString(HEADER_TYPE, JWT_ACCESS_TOKEN)
+        val authorization =
+            if (headerType == JWT_REFRESH_TOKEN)
+                authHolder.getRefreshAuthorization()
+            else
+                authHolder.getAccessAuthorization()
```

Comment:
No, no, no. How does it refresh the token when it is expired?

See the ansers here https://stackoverflow.com/questions/22450036/refreshing-oauth-token-using-retrofit-without-modifying-all-calls

Reply:
Good approach. :+1: